app/shop.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int, int],
+            products: dict[str, float],
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @classmethod
+    def from_dict(
+            cls,
+            shop_dict: dict,
+    ) -> Shop:
+        return cls(
+            shop_dict.get("name"),
+            shop_dict.get("location"),
+            shop_dict.get("products")
+        )
+
+    def service(self, customer_name: str, desired_products: dict) -> float:
+        products_to_buy = {}
+        total_cost = 0
+
+        for product, quantity in desired_products.items():
+            price = quantity * self.products[product]
+            total_cost += price
+
+            if float(price).is_integer():
+                price = int(price)
+            else:
+                round(price, 1)
```

Comment:
The line `round(price, 1)` does not modify `price` because the `round` function returns a new value. You should assign the result back to `price` to ensure it is rounded to one decimal place.